src/components/modals/TransferPDNTModal/TransferPDNTModal.tsx

```diff
@@ -0,0 +1,260 @@
+          >
+            <span
+              className="text-small white bold"
+              style={{
+                textShadow: '2px 2px 2px rgb(0,0,0)',
+              }}
+            >
+              Transfer ANT:&nbsp;{state.ticker ?? state.name}
+            </span>
+            <span className="flex faded text center">
+              Contract ID:&nbsp;
+              <CopyTextButton
+                copyText={pdntId ? pdntId.toString() : ''}
+                displayText={`${pdntId!.toString().slice(0, isMobile ? 6 : 0)}${
+                  isMobile ? '...' : ''
+                }${pdntId!.toString().slice(isMobile ? -6 : 0)}`}
+                size={'70%'}
+                wrapperStyle={{
+                  fontStyle: 'bold',
+                  fontSize: '10px',
+                  justifyContent: 'center',
+                  alignItems: 'center',
+                }}
+                position="relative"
+              />
+            </span>
+            <button className="modal-close-button" onClick={() => showModal()}>
+              <CloseIcon
+                width="20px"
+                height={'20px'}
+                fill="var(--text-white)"
+              />
+            </button>
+          </div>
+          {/** modal body - condition render edit or transfer */}
+
+          <div
+            className="flex flex-column"
+            style={{
+              width: '90%',
+              height: '100%',
+              gap: '1em',
+            }}
+          >
+            {/* todo: show 'transfer between accounts' option */}
+
+            <ValidationInput
+              inputClassName="data-input center"
+              inputCustomStyle={
+                isValidAddress && toAddress
+                  ? {
+                      border: '2px solid var(--success-green)',
+                    }
+                  : !isValidAddress && toAddress
+                  ? {
+                      border: '2px solid var(--error-red)',
+                    }
+                  : !isValidAddress && !toAddress && {}
+              }
+              validationListStyle={{
+                gap: isMobile ? '0.5em' : '1em',
+                justifyContent: 'center',
+                alignItems: 'center',
+              }}
+              wrapperClassName="flex flex-column center"
+              wrapperCustomStyle={{ gap: isMobile ? '0.5em' : '1em' }}
+              placeholder="Enter recipients address"
+              maxLength={43}
+              value={toAddress}
+              setValue={setToAddress}
+              validityCallback={(validity: boolean) =>
+                setIsValidAddress(validity)
+              }
+              showValidationChecklist={true}
+              validationPredicates={{
+                [VALIDATION_INPUT_TYPES.ARWEAVE_ID]: (id: string) =>
+                  arweaveDataProvider.validateArweaveId(id),
+              }}
+            />
+            {getAssociatedNames(pdntId!, pdnsSourceContract.records).length ? (
```

Comment:
why force unwrap here?

also, we're calling this function multiple times, maybe move it above

Reply:
moved